dev/sensor/htu21/export/dev/sensor/htu21.hpp

```diff
@@ -0,0 +1,181 @@
+#ifndef __HTU21_HPP__
+#define __HTU21_HPP__
+
+namespace sensor {
+
+template <class i2c_dev>
+class htu21 {
+public:
+    htu21() = delete;
+    ~htu21() = delete;
+
+    //! \brief Inits sensor and underlying I2C platform bus.
+    //! \retval Status of operation.
+    //!
+    static ecl::err init();
+
+    //! \brief Performs soft reset of the sensor.
+    //!
+    static void soft_reset();
+
+    //! \brief Reads sensor user register.
+    //! \details Allows to get useful information about sensor state
+    //! \retval User register content.
+    //!
+    static uint8_t read_user_register();
+
+    //! \brief Writes to the sensor user register.
+    //! \details Allows to configure sensor resolution.
+    //!
+    static void write_user_register(uint8_t value);
+
+    //! \brief Reads raw temperature sample from sensor.
+    //! \details Sample should be processed to receive physical value.
+    //! \retval Raw temperature sample.
+    //!
+    static uint16_t get_sample_temperature();
+
+    //! \brief Reads raw relative humidity sample from sensor.
+    //! \details Sample should be processed to receive physical value.
+    //! \retval Raw relative humidity sample.
+    //!
+    static uint16_t get_sample_humidity();
+
+    //! \brief Reads sample from sensor and converts to a physical value.
+    //! \retval Temperature in C degree.
+    //!
+    static float get_temperature();
```

Comment:
That's dangerous. Not all hardaware have hardware floating point unit, and not all HW works fast. Let's use an integer, interpreted as real number. Say, value of 12345 will mean 12.345 degree in Celsius.